### ISSUE TITLE
Can access attribute "__args__" for local class "TypeAliasType", but Cannot access attribute "__args__" for imported class "TypeAliasType"

### ISSUE THREAD

**Issue by hunterhogan** [microsoft/pyright#11484]
Type: <b>Bug</b>

The local version of the code does not produce diagnostics, but the imported version does.

```python
from __future__ import annotations

from soundfile import dtype_str
from typing import Literal, TypeAlias

dtype_str_local: TypeAlias = Literal['float64', 'float32', 'int32', 'int16']

qq = dtype_str_local.__args__
ww = dtype_str.__args__
```

<details><summary>Cannot access attribute "__args__" for class "TypeAliasType"</summary>
<p>

```json
[{
	"resource": "/C:/apps/hunterHearsPy/ee.py",
	"owner": "Pylance",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot access attribute \"__args__\" for class \"TypeAliasType\"\n  Attribute \"__args__\" is unknown",
	"source": "Pylance",
	"startLineNumber": 9,
	"startColumn": 16,
	"endLineNumber": 9,
	"endColumn": 24,
	"modelVersionId": 34,
	"origin": "extHost1"
}]
```

</p>
</details> 

Which I assume necessarily leads to 

<details><summary>Type of "__args__" is unknown</summary>
<p>

```json
[{
	"resource": "/C:/apps/hunterHearsPy/ee.py",
	"owner": "Pylance",
	"code": {
		"value": "reportUnknownMemberType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportUnknownMemberType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type of \"__args__\" is unknown",
	"source": "Pylance",
	"startLineNumber": 9,
	"startColumn": 6,
	"endLineNumber": 9,
	"endColumn": 24,
	"modelVersionId": 34,
	"origin": "extHost1"
}]
```

</p>
</details> 

The Python interpreter can see the `__args__` attribute.

<details><summary>Details</summary>
<p>

```pycon
(hunterhearspy) C:\apps\hunterHearsPy>ver

Microsoft Windows [Version 10.0.26200.8655]

(hunterhearspy) C:\apps\hunterHearsPy>py
Python 3.14.5 (tags/v3.14.5:5607950, May 10 2026, 10:43:50) [MSC v.1944 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
Ctrl click to launch VS Code Native REPL
>>> from soundfile import dtype_str
>>> dtype_str.__dir__()
['__origin__', '__str__', '__or__', '__init_subclass__', '_inst', '__args__', '__ror__', 'copy_with', '__mro_entries__', '__getitem__', '__subclasshook__', '_getitem', '__static_attributes__', '__doc__', '__setattr__', '__le__', '__gt__', '_make_substitution', '__instancecheck__', '__ge__', '_name', '__getattribute__', '__format__', '__class__', '__dir__', '_determine_new_args', '__reduce_ex__', '__weakref__', '__new__', '__ne__', '__repr__', '__subclasscheck__', '__getstate__', '__init__', '__eq__', '__firstlineno__', '__delattr__', '__sizeof__', '__slots__', '__call__', '__getattr__', '__hash__', '__parameters__', '__module__', '__lt__', '__iter__', '__dict__', '__reduce__']
>>> dtype_str.__args__ 
('float64', 'float32', 'int32', 'int16')
>>> type(dtype_str.__args__)
<class 'tuple'>
``` 

</p>
</details> 

---
Extension version: 2026.2.104
VS Code version: Code - Insiders 1.125.0-insider (d7ce0ce923f97583f7e03f9c17a149b1601090a3, 2026-06-11T13:27:59+10:00)
OS version: Windows_NT x64 10.0.26200
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 5700U with Radeon Graphics          (16 x 1797)|
|GPU Status|2d_canvas: enabled<br>GPU0: VENDOR= 0x1002, DEVICE=0x164c [AMD Radeon(TM) Graphics], DRIVER_VENDOR=AMD, DRIVER_VERSION=31.0.21925.1001 *ACTIVE*<br>GPU1: VENDOR= 0x1414, DEVICE=0x008c [Microsoft Basic Render Driver], DRIVER_VERSION=10.0.26100.8521<br>Machine model name: <br>Machine model version: <br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>trees_in_viz: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|12.90GB (3.10GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by rchiodo** [comment]
What does the imported version look like? I can't reproduce it with this:

<img width="770" height="195" alt="Image" src="https://github.com/user-attachments/assets/0cbb9263-2af0-426d-96d7-d9a5fb0e2bbe" />

**Comment by hunterhogan** [comment]
Now, it's working?!

<img width="716" height="202" alt="Image" src="https://github.com/user-attachments/assets/e4c0288e-6c04-43c8-9fc0-7641314beeda" />

I updated VS Code, but that is probably the only change.

> Version: 1.125.0-insider (user setup)
Commit: d420356ab9998a711c36d7c51a2d7d2c91c73a87
Date: 2026-06-11T16:58:45Z
Electron: 42.2.0
ElectronBuildId: 14159160
Chromium: 148.0.7778.97
Node.js: 24.15.0
V8: 14.8.178.14-electron.0
OS: Windows_NT x64 10.0.26200

In the original post, the folded details have the diagnostics, but it is suddenly working!

To be frank, for a couple of months, I've suspected that something else is wrong with VS Code; my OS; my configuration of VS Code, Python, or my OS; or my hardware, possibly. But despite tons of troubleshooting and reseating every connection inside and outside my computer, I haven't narrowed it down--and I haven't convinced myself that there _isn't_ a common cause of some weird issues I'm having. 

(Have you seen weird, random problems when Serena MCP is installed?)

In any event, I don't think this is a problem with `.__args__` specifically. You might remember a weird issue with the language server only suggesting deprecated classes from typing instead of collections.abc. _IF_ there is a Pylance/Pyright angle to the above one-off/intermittent problem, _then_ I suspect it would be related to that problem.

(Dude, I'm sorry my writing is terrible. I just woke up--no coffee. But if I don't write these details now, I know I will never write these details. You, @rchiodo, are an _outstanding_ troubleshooter, so I'm trying to do a brain dump of my observations. My hope is that in some future issue, one of these details will be useful.)

Back to the old issue. I don't know if VS Code/Pylane/Pyright fixed the problem, but I have mitigated the problem with two actions. First, I used to have a symlink in {workspace root}/typings to the contents of hunterhogan/stubFileNotFound/stubs. I firmly believe that the quantity of stubs was too much for Pylance/Pyright. I changed that tactic: I don't symlink the entire stubs directory. I make a {workspace root}/typings directory, then I _ad hoc_ symlink 

```cmd
@ECHO OFF
PUSHd typings

MKLINK /D "%~1" "\apps\StubFileNotFound\stubs\%~1"

POPd
```

to specific packages. I believe that helps. It happens to be that I have a custom stub for `soundfile` and the associated symlink, and it happens to be that I spent many hours struggling with a typing issue (still not fully resolved) that included nine overloads in soundfile. So, it's possible _too much_ typing information about soundfile had a bad side effect. (My guestimate: 1-2% probability.)

The second mitigation: system-wide install of Ruff (uv tool), Ruff extension, and Ruff LSP running with Pylance LSP. One of the two LSP usually recommends the correct import path. For multiple months, I don't think having two LSP has caused problems. But recently, I've been starting to wonder if a change in Ruff is now leading to random problems.

/brain_dump

If you want to keep this open, you can. From my perspective, it's either intermittent or a side-effect of a different problem.

Thanks for your time.

**Comment by rchiodo** [comment]
Okay thanks for responding.
